### PR TITLE
Browser support.

### PR DESCRIPTION
**Pull request by alexjeffburke**
This PR introduces browser support and rewires the test suite to ensure it is exercised.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const puppeteer = require("puppeteer");
+process.env.CHROME_BIN = puppeteer.executablePath();
```
Comment (sunesimonsen) [inline karma.conf.js]: Why is this necessary? Can't we just ask for a Chrome instance:

https://github.com/unexpectedjs/unexpected/blob/master/.travis.yml#L34-L35

Comment (sunesimonsen) [reply karma.conf.js]: It would be nice to remove puppeteer.

Comment (alexjeffburke) [reply karma.conf.js]: Oh wow, just saw the Chrome addon thing and had no idea you could ask for it! Let's do that :-)
